src/printer.rs

```diff
@@ -430,6 +430,15 @@ impl<'a> Printer for InteractivePrinter<'a> {
                     .decode(line_buffer, DecoderTrap::Replace)
                     .map_err(|_| "Invalid UTF-16BE")?,
                 _ => String::from_utf8_lossy(line_buffer).to_string(),
+            };
+            // Remove byte order mark from the first line if it exists
+            if line_number == 1 {
+                match line.strip_prefix('\u{feff}') {
+                    Some(stripped) => stripped.to_string(),
```

Comment:
I wonder if there is an alternative implementation where we do not have to allocate a new string? by operating on `&str` instead of `String`.

But .. this is only relevant for files starting with a BOM. So maybe it's not the most urgent question.